### ISSUE TITLE
Channel Discovery

### ISSUE THREAD

**Issue by NigeWarren** [Tesco/mewbase#179]
Just writing up this  https://github.com/Tesco/mewbase/wiki

It occurred that the identity of channels is assumed to be known at Service compile time however it may be useful in certain services to be able to discover channels at runtime. Should there be some form of channel discovery and which API should this be on?

If we were to introduce Channel base security then it makes sense for a service, given an identity to be able to only discover the Channels that it can Publish and/or Subscribe to.

Which implies that in order to do this we need the notion of Service Identity which should probably follow the FQCN strategy adopted for Channels.

**Comment by kings13y** [comment]
Wondering if there's anything in the CNCF cloudevents spec that we could use here ? 
https://github.com/cloudevents/spec
tldr; thinking of something like WSDL/WADL etc for Channels, event streams and event payloads.. will have a think about it..
